src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,315 @@
+      return s;
+    }
+    UUID id = UUID.fromString(termNode.getTerm()); // so IllegalArgumentException is thrown
+    String pgTerm = "'" + id + "'";
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  /**
+   * Convert CQL term to Postgres term - exact - without C style escapes in result.
+   * <p> Double backslash is converted to backslash. Postgres quotes (') are escaped.
+   * Otherwise things are passed through verbatim.
+   * </p>
+   * @param termNode termNode which includes term and relation.
+   * @return Postgres term without C style escapes.
+   */
+  static String cqlTermToPgTermExact(CQLTermNode termNode) {
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      if (c == '\\' && backslash) {
+        backslash = false;
+      } else {
+        pgTerm.append(c);
+        if (c == '\'') {
+          pgTerm.append('\''); // important to avoid SQL injection
+        }
+        backslash = c == '\\';
+      }
+    }
+    return pgTerm.toString();
+  }
+
+  /**
+   * CQL full text term to Postgres term.
+   * @see <a href="https://www.postgresql.org/docs/13/sql-syntax-lexical.html">
+   *   String Constants section</a>
+   *
+   * <p>At this stage masking is unsupported and rejected.</p>
+   * @param termNode which includes term and relation.
+   * @return Postgres term.
+   */
+  static String cqlTermToPgTermFullText(CQLTermNode termNode) {
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      // handle the CQL specials *, ?, ^, \\, rest are passed through as is
+      if (c == '*') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op * unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '?') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op ? unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '^') {
+        if (!backslash) {
+          throw new IllegalArgumentException("Anchor op ^ unsupported for: " + termNode.toCQL());
+        }
+      } else if (c != '\\' && backslash) {
+        pgTerm.append('\\');
+      }
+      if (c == '\\' && !backslash) {
+        backslash = true;
+      } else {
+        pgTerm.append(c);
+        if (c == '\'') {
+          pgTerm.append(c);
+        }
+        backslash = false;
+      }
+    }
+    return pgTerm.toString();
+  }
+
+  String handleTypeText(PgCqlField field, CQLTermNode termNode, boolean fullText) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String base = termNode.getRelation().getBase();
+    if (fullText) {
+      fullText = "=".equals(base) || "all".equals(base);
```

Comment:
plainto_tsquery implements the "all" search, each word of the query string exists somewhere (any order, any other words may be between the query words).

RMB implements "=" as "adj": A substring phrase match - all words of the query string exist consecutively in that order (ignoring differences in whitespace and punctuation).

RMB follows the recommendation of the CQL spec:
> =
> This is the default relation, and the server can choose any appropriate relation […]
> For a string term, either 'adj' or '==' as appropriate for the index and term. 

https://www.loc.gov/standards/sru/cql/contextSets/theCqlContextSet.html#relations

This new code implements "=" as "all".

This deviates from RMB and the CQL spec.

Is this intended?

If yes it should be mentioned in a README (or similar file) with a big warning so that clients can code around this special case in mod-eusage-reports.

Reply:
This was intended to produce a small implementation since that is what Postgres supports out of the box.